builtin/providers/spotinst/resource_spotinst_aws_group.go

```diff
@@ -0,0 +1,1837 @@
+package spotinst
+
+import (
+	"bytes"
+	"crypto/sha1"
+	"encoding/base64"
+	"encoding/hex"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/spotinst/spotinst-sdk-go/spotinst"
+)
+
+func resourceSpotinstAwsGroup() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSpotinstAwsGroupCreate,
+		Read:   resourceSpotinstAwsGroupRead,
+		Update: resourceSpotinstAwsGroupUpdate,
+		Delete: resourceSpotinstAwsGroupDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+			},
+
+			"capacity": &schema.Schema{
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: false,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"minimum": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: false,
+						},
+
+						"maximum": &schema.Schema{
+							Type:     schema.TypeInt,
+							Optional: true,
+							ForceNew: false,
+						},
+
+						"target": &schema.Schema{
+							Type:     schema.TypeInt,
+							Required: true,
+							ForceNew: false,
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(fmt.Sprintf("%d-", m["minimum"].(int)))
+					buf.WriteString(fmt.Sprintf("%d-", m["maximum"].(int)))
+					buf.WriteString(fmt.Sprintf("%d-", m["target"].(int)))
+					return hashcode.String(buf.String())
+				},
+			},
+
+			"strategy": &schema.Schema{
+				Type:     schema.TypeSet,
```

Comment:
What's the best practice? I'm not sure why the Set func is necessary in this case, or even for the `capacity` schema.